dropshot/src/api_description.rs

```diff
@@ -652,6 +672,53 @@ impl<Context: ServerContext> ApiDescription<Context> {
         // Sort the tags for stability
         openapi.tags.sort_by(|a, b| a.name.cmp(&b.name));
 
+        // Error return types are a bit complex.
+        //
+        // We anticipate that multiple API endpoints may return the same error
+        // type, and in that case, we would prefer for the error type to be
+        // defined once in the OpenAPI document and referenced by all endpoints
+        // that return it, so that client code generators can generate a single
+        // type that's used for all endpoints that return a given error type.
+        // However, there's a catch: the `schemars::JsonSchema` trait's `derive`
+        // macro generates the name of the Rust type as the name of the schema,
+        // *not* the fully qualified name (i.e. the type name and module path).
+        // This means that if one API endpoint returns a `foo::Error`, and
+        // another one returns a `bar::Error`, and we use the schema name
+        // according to schemars as the key for referencing the schema, both
+        // endpoints will return the same schema in the OpenAPI document, but
+        // the actual Rust types may serialize according to totally different
+        // schemas. Therefore, we must ensure that if two error return types
+        // have the same Rust type name, we instead refer to their schemas by
+        // the fully-qualified module path, rather than just the type name.
+        // However, this is ugly, so we would prefer to only do it when it's
+        // necessary to disambiguate stuff.
```

Comment:
Doesn't this apply to regular types, too?  I think you run into the same thing if you have two ordinary return types (or types used in types) whose schema name is the same (often, the Rust type basename is the same, but they're in different modules).  Are errors different here?